### ISSUE TITLE
Core dumped - Index >= 0 && index < size()

### ISSUE THREAD

**Issue by juliandwillett** [rgcgithub/regenie#430]
I have been getting the following error in Regenie, albeit in an analysis where I included sex chromosomes, but did not correct the ploidy. 

```
* making predictions...regenie_v3.2.8.gz_x86_64_Linux: /home/s.joelle.mbatchou/software/regenie/external_libs/eigen-3.4.0/Eigen/src/Core/DenseCoeffsBase.h:427: Eigen::DenseCoeffsBase<Derived, 1>::Scalar& Eigen::DenseCoeffsBase<Derived, 1>::operator()(Eigen::Index) [with Derived = Eigen::Array<int, -1, 1>; Eigen::DenseCoeffsBase<Derived, 1>::Scalar = int; Eigen::Index = long int]: Assertion `index >= 0 && index < size()' failed.
Aborted (core dumped)
```

Everything in the pipeline works until it tries to make predictions after outputting the phenotype likelihood/N values. I have already isolated the autosomes, although would appreciate guidance on avoiding this from repeating, due to cloud computing costs. My code is below:

```
./regenie_v3.2.8.gz_x86_64_Linux \
  --step 1 \
  --bgen arrays_autosomes_post_qc_pruned_onlysrwgs.bgen \
  --sample arrays_autosomes_post_qc_pruned_onlysrwgs.sample \
  --phenoFile regenie_pheno.txt \
  --covarFile regenie_covars.txt \
  --bt \
  --bsize 1000 \
  --out aou_step1_rg_array \
  --lowmem \
  --lowmem-prefix tmp_rg 
```

**Comment by juliandwillett** [comment]
This did not occur when I only operated on chromosome 1.

**Comment by juliandwillett** [comment]
I excluded the sex chromosomes from my analysis (which I did not previously adjust for ploidy), and my analysis worked out. While there were other considerations, such as having a different number of individuals in my phenotype, covariate, and input bgen files, this did not interfere with success of the algorithm.

**Comment by Sabor117** [comment]
Hi there, can I ask what the solution to this issue was?

I have recently conducted a run of REGENIE on chromosomes 1-22 without any issue but then I needed to go back to re-run it incorporating the X chromosome and when I did this I got exactly the same error:

`making predictions...regenie_v3.2.8.gz_x86_64_Centos7_mkl: /src/regenie/external_libs/eigen-3.4.0/Eigen/src/Core/DenseCoeffsBase.h:427: Eigen::DenseCoeffsBase<Derived, 1>::Scalar& Eigen::DenseCoeffsBase<Derived, 1>::operator()(Eigen::Index) [with Derived = Eigen::Array<int, -1, 1>; Eigen::DenseCoeffsBase<Derived, 1>::Scalar = int; Eigen::Index = long int]: Assertion `index >= 0 && index < size()' failed.`

In the REGENIE tutorial it mentions that "males should be coded as diploid so that their genotypes are 0/2 (this is done automatically for BED..." and as I am using a combined set of bed/bim/fam files containing all chromosomes (including X), I had assumed these files would just be inherently okay without performing any additional work on them.

Do I need to do something else with Plink in order to change the coding of males? Or was there a different solution? (I am currently re-running one of these GWAS with the `--debug` and `--verbose` arguments to try and get more information about the error.
